File: frontend/src/types/product.ts
export interface Product {
  id: number;
  name: string | null;
  barcode: string | null;
  category: string | null;
  manufacturer_name: string | null;
  manufacturer_address: string | null;
  net_quantity: string | null;
  mrp: number | null;
  image_url: string | null;
  created_at: string;
  scan_count?: number;
  last_scan_at?: string | null;
  last_compliance_status?: string | null;
  last_compliance_score?: number | null;
  scans?: {
    scan_id: string;
    compliance_status: string;
    compliance_score: number | null;
    total_violations: number;
    created_at: string;
  }[];
}

File: backend/app/schemas/product.py
"""Product schemas."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class ProductResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: Optional[str] = None
    barcode: Optional[str] = None
    category: Optional[str] = None
    manufacturer_name: Optional[str] = None
    manufacturer_address: Optional[str] = None
    net_quantity: Optional[str] = None
    mrp: Optional[float] = None
    image_url: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None


class ProductScanSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    scan_id: str
    compliance_status: str
    compliance_score: Optional[float] = None
    total_violations: int = 0
    created_at: datetime


class ProductDetailResponse(ProductResponse):
    scans: list[ProductScanSummary] = []


class ProductListResponse(BaseModel):
    total: int
    items: list[ProductResponse]
